src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -333,26 +344,29 @@ internal SourceText TryReadFileContent(CommandLineSourceFile file, IList<Diagnos
                             if (regexes[sectionIndex].IsMatch(relativePath))
                             {
                                 var section = config.NamedSections[sectionIndex];
-                                addDiagnosticOptions(section, optionsBuilder, config.PathToFile);
+                                addOptions(section, treeOptionsBuilder, ref analyzerOptions, config.PathToFile);
                             }
                         }
                     }
                 }
 
-                // Avoid extra allocations by using null.
-                allDiagnosticOptions.Add(optionsBuilder.Count > 0 ? optionsBuilder.ToImmutable() : null);
-                optionsBuilder.Clear();
+                allDiagnosticOptions.Add(
+                    // Avoid extra allocations by using null.
```

Comment:
❔ How is `null` better than `ImmutableDictionary<string, ReportDiagnostic>.Empty`?

Reply:
`Empty` has the wrong comparer. The API allows for `null` and uses a cached singleton.